src/HttpHeader.cc

```diff
@@ -1464,6 +1448,20 @@ HttpHeaderEntry::parse(const char *field_start, const char *field_end, const htt
         }
     }
 
+    /* RFC 7230 section 3.2:
+     *
+     *  header-field   = field-name ":" OWS field-value OWS
+     *  field-name     = token
+     *  token          = 1*TCHAR
+     */
+    for (const char *pos = field_start; pos < name_end; ++pos) {
+        if (!CharacterSet::TCHAR[*pos]) {
+            debugs(55, 2, "ignoring header with invalid characters found in field-name (" <<
```

Comment:
AFAICT, this `return nullptr` code does not just quietly ignore the header (field). It effectively rejects the whole message, with a WARNING in cache.log! I suggest not documenting the future high-level outcome of our low-level observations and using "found" instead of "ignoring" in this debugs() line:

```suggestion
            debugs(55, 2, "found header with invalid characters in field-name (" <<
```

Reply:
Done in 11bfb66